WIP - MTA-4637: Pattern matching docs need urgent improvements (#92)
migtools/mta-documentation: docs/topics/yaml-rulesets.adoc

// Module included in the following assemblies:
//
// * docs/rules-development-guide/master.adoc

:_mod-docs-content-type: REFERENCE
[id="yaml-rulesets_{context}"]
= Rulesets

A set of rules forms a ruleset. {ProductShortName} does not require every rule file to belong to a ruleset, but you can use rulesets to group multiple rules that achieve a common goal and to pass the rules to the rules engine.

> You can create a ruleset by placing one or more YAML rules in a directory and creating a `ruleset.yaml` file at the directory root. When you pass this directory as input to the {ProductShortName} {CLIName} using the `--rules` option, all rules in this directory are treated as a part of the ruleset defined by the `ruleset.yaml` file.

Reviewer: ```suggestion
You can create a ruleset by placing one or more YAML rules in a directory and creating a `ruleset.yaml` file at the directory root. When you pass this directory as input to the {ProductShortName} {CLIName} by using the `--rules` option, all rules in this directory are treated as a part of the ruleset defined by the `ruleset.yaml` file.
```

The `ruleset.yaml` file stores the metadata of the ruleset.

[source,yaml]
----
name: "Name of the ruleset" # <1>
description: "Description of the ruleset"
labels: #  <2>
  - key=val
----
<1> The name must be unique within the provided rulesets.
<2> Ruleset labels are inherited by all rules that belong to the ruleset.

> To perform any application analysis, run the following command and replace the following:
> * `<application_to_analyze>` with your application
> *  `<output_dir>` with the directory of your choice
> *  `<custom_rule_dir>` with the custom rulesets file
> 
> [source,terminal]
> ----
> $ mta-cli analyze --input=<application_to_analyze> --output=<output_dir> --rules=<custom_rule_dir> --enable-default-rulesets=false
> ----

Reviewer: ```suggestion
To perform any application analysis, enter:

[source,terminal]
----
$ mta-cli analyze --input=<application_to_analyze> --output=<output_dir> --rules=<custom_rule_dir> --enable-default-rulesets=false
----

*  Replace `<application_to_analyze>` with the name of your application.
*  Replace `<output_dir>` with the directory of your choice.
*  Replace `<custom_rule_dir>` with the custom rulesets file.
```

> On initiation, the mta-cli tool determines the type of application and the  provider needed for analysis. It then starts the provider in a container that has the required dependencies and tools. Finally, the provider uses the analyzer to execute a series of rulesets to analyze the source code.

Reviewer: ```suggestion
On initiation, the mta-cli tool determines the type of application and the provider needed for analysis. It then starts the provider in a container that has the required dependencies and tools. Finally, the provider uses the analyzer to execute a series of rulesets to analyze the source code.
```
